src/ComboBox/utils.js

```diff
@@ -52,10 +52,19 @@ function prefixOptions( options = [], prefix )
             };
         }
 
-        return {
+        const flatOptions = {
+            text : option,
+            id   : addPrefix( option, prefix ),
+        };
+        const structuredOptions = {
             ...option,
             id : addPrefix( option.id, prefix ),
         };
+
+        const returnedOptions = typeof option !== 'string' ?
+            structuredOptions : flatOptions;
+
+        return returnedOptions;
     } );
 }
```

Comment:
these changes wouldn't be necessary as we should pass already a "Normalized Options" array to `prefixOptions()`